### ISSUE TITLE
CI: CUDA Python test is broken

### ISSUE THREAD

**Issue by kmaehashi** [cupy/cupy#6854]
CUDA Python CI is broken. It seems CUDA 11.5 & CUDA Python 11.7.1 & texture memory issue.

https://ci.preferred.jp/cupy.linux.cuda11x-cuda-python/105617/

```
01:27:04.155812 STDOUT 1589]	FAILED cupy_tests/core_tests/test_userkernel.py::TestElementwiseKernelTexture_param_0_{dimensions=(64, 0, 0)}::test_texture_input	
01:27:04.155819 STDOUT 1589]	FAILED cupy_tests/core_tests/test_userkernel.py::TestElementwiseKernelTexture_param_1_{dimensions=(64, 32, 0)}::test_texture_input	
01:27:04.155823 STDOUT 1589]	FAILED cupy_tests/core_tests/test_userkernel.py::TestElementwiseKernelTexture_param_2_{dimensions=(64, 32, 19)}::test_texture_input	
01:27:04.155828 STDOUT 1589]	FAILED cupy_tests/cuda_tests/test_texture.py::TestTexture::test_fetch_float_texture[_param_0_{dimensions=(64, 0, 0), mem_type='CUDAarray', target='object'}]	
01:27:04.155834 STDOUT 1589]	FAILED cupy_tests/cuda_tests/test_texture.py::TestTexture::test_fetch_float_texture[_param_6_{dimensions=(64, 32, 0), mem_type='CUDAarray', target='object'}]	
01:27:04.155838 STDOUT 1589]	FAILED cupy_tests/cuda_tests/test_texture.py::TestTexture::test_fetch_float_texture[_param_12_{dimensions=(64, 32, 19), mem_type='CUDAarray', target='object'}]	
01:27:04.155844 STDOUT 1589]	FAILED cupy_tests/cuda_tests/test_texture.py::TestTextureVectorType::test_fetch_float4_texture[_param_0_{target='object'}]	
01:27:04.155886 STDOUT 1589]	= 7 failed, 98674 passed, 3634 skipped, 1043 deselected, 88 xfailed, 2569 warnings in 3444.73s (0:57:24) =	
```

**Comment by leofang** [comment]
CUDA Python 11.7.1 needs to be built with CUDA 11.7.1, maybe this is the reason?

**Comment by kmaehashi** [comment]
Oh, is that so? I thought CUDA Python 11.X would work with CUDA 11.0~11.X.

https://github.com/nvidia/cuda-python
> CUDA Toolkit 11.0 to 11.7

**Comment by leofang** [comment]
CUDA Python 11.7.1 needs to see the same version of CUDA headers at the build time. At runtime the compatibility remains.

**Comment by leofang** [comment]
(It was done to fix https://github.com/NVIDIA/cuda-python/issues/22.)

**Comment by leofang** [comment]
I took a quick look at the failures, looks like they're all due to results mismatching at runtime, and they're all about texture, perhaps there's something else going on...

**Comment by kmaehashi** [comment]
> I took a quick look at the failures, looks like they're all due to results mismatching at runtime, and they're all about texture, perhaps there's something else going on...

Seems so. CI uses `cuda-python` binary package so the source build restrictions do not apply.

**Comment by anaruse** [comment]
I built CuPy (v11) with the following combinations and tested `cupy_tests/core_tests/test_userkernel.py` and `cupy_tests/cuda_tests/test_texture.py`. The error occurred only with the combination of CUDA 11.5.0 and CUDA-python 11.7.1. 

For CUDA 11.5, is it possible to use, for example, CUDA-python 11.7.0 as a workaround?

| CUDA | CUDA-python | test results |
:---:|:---:|:---:
| 11.5.0 | 11.5.0 | OK |
| 11.5.0 | 11.6.0 | OK |
| 11.5.0 | 11.7.0 | OK |
| 11.5.0 | 11.7.1 | **NG** |
| 11.6.2 | 11.5.0 | OK |
| 11.6.2 | 11.6.0 | OK |
| 11.6.2 | 11.7.0 | OK |
| 11.6.2 | 11.7.1 | OK |
| 11.7.0 | 11.5.0 | OK |
| 11.7.0 | 11.6.0 | OK |
| 11.7.0 | 11.7.0 | OK |
| 11.7.0 | 11.7.1 | OK |
| 11.7.1 | 11.5.0 | OK |
| 11.7.1 | 11.6.0 | OK |
| 11.7.1 | 11.7.0 | OK |
| 11.7.1 | 11.7.1 | OK |

**Comment by kmaehashi** [comment]
Thanks for exhaustive testing, @anaruse! Yes, we have no reason to stick with CUDA 11.5. Will bump the CUDA version in #7022.

**Comment by anaruse** [comment]
The CUDA python team's investigation has revealed that this is not directly due to CUDA python, but is essentially due to backward compatibility issues with the texture related APIs in CUDA 11.5/11.6. This backward compatibility issue has been resolved in CUDA 11.7, so it would be helpful if you could use CUDA 11.7 to test CUDA python.
